tests/jballerina-unit-test/src/test/resources/test-src/functions/functions_with_default_parameters.bal

```diff
@@ -300,6 +300,20 @@ function value2(record {|int a; int b;|} x) returns int {
     return x.a + x.b;
 }
 
+type F1 function() returns int;
+
+function execute(F1 f) returns int {
+    return f();
+}
+
+function baz(int a, F1 b = function() returns int { return a + 1; }) returns int {
```

Comment:
Just to be sure shall we also test a case where second function uses the first argument as a default value as well. Something like 
```
function foo(int a, function(int x = a) returns int b = function(int y) returns int { return y + a; }) returns int {
    return a + b();
}
```

Reply:
According to the spec this should be a compile time error(https://github.com/ballerina-platform/ballerina-lang/issues/41871).